modules/gitops-argo-cd-dynamic-scaling.adoc

```diff
@@ -0,0 +1,112 @@
+. Select the Argo CD instance name for which you want to enable dynamic scaling of shards, for example `openshift-gitops`.
+. Click on the *YAML* tab, and then edit and configure the `spec.controller.sharding` properties as follows:
++
+.Example Argo CD YAML file with dynamic scaling enabled
+[source,yaml]
+----
+apiVersion: argoproj.io/v1beta1
+kind: ArgoCD
+metadata:
+  name: openshift-gitops
+  namespace: openshift-gitops
+spec:
+  controller:
+    sharding:
+      dynamicScalingEnabled: true <1>
+      minShards: 1 <2>
+      maxShards: 3 <3>
+      clustersPerShard: 1 <4>
+----
+<1> Set `dynamicScalingEnabled` to `true` to enable dynamic scaling.
+<2> Set `minShards` to the minimum number of shards that you want to have. The value must be set to `1` or greater.
+<3> Set `maxShards` to the maximum number of shards that you want to have. The value must be greater than the value of `minShards`.
+<4> Set `clustersPerShard` to the number of clusters that you want to have per shard. The value must be set to `1` or greater.
+
+. Click *Save* to save the changes.
+
+* To enable dynamic scaling of shards using the command-line interface, complete the following steps:
+. Log in to the cluster using the `oc` command-line tool as a user with `cluster-admin` privileges.
+. Enable dynamic scaling by running the following command:
++
+[source,terminal]
+----
+$ oc patch argocd <argocd_instance> -n <namespace> --type=merge --patch='{"spec":{"controller":{"sharding":{"dynamicScalingEnabled":true,"minShards":<value>,"maxShards":<value>,"clustersPerShard":<value>}}}}'
+----
++
+.Example command
+[source,terminal]
+----
+$ oc patch argocd openshift-gitops -n openshift-gitops --type=merge --patch='{"spec":{"controller":{"sharding":{"dynamicScalingEnabled":true,"minShards":1,"maxShards":3,"clustersPerShard":1}}}}' <1>
+----
++
+<1> The example command enables dynamic scaling for the `openshift-gitops` Argo CD instance in the `openshift-gitops` namespace, and sets the minimum number of shards to `1`, the maximum number of shards to `3`, and the number of clusters per shard to `1`.
++
+.Example output
+[source,terminal]
+----
+argocd.argoproj.io/openshift-gitops patched
+----
+
+. Verify that dynamic scaling is enabled by running the following command: 
++
+[source,terminal]
+----
+$ oc get argocd <argocd_instance> -n <namespace> -o jsonpath='{.spec.controller.sharding}'
+----
++
+.Example command 
+[source,terminal]
+----
+$ oc get argocd openshift-gitops -n openshift-gitops -o jsonpath='{.spec.controller.sharding}'
+----
++
+.Example output when dynamic scaling of shards is enabled
+[source,terminal]
+----
+{"dynamicScalingEnabled":true,"minShards":1,"maxShards":3,"clustersPerShard":1}
+----
+
+.. Alternatively, you can verify that dynamic scaling is enabled by checking the configured `spec.controller.sharding` properties in the configuration `YAML` file of the Argo CD instance on the {OCP} web console.
```

Comment:
Move the verification steps to its respective sections. So the first one stays under CLI, and the second one will no longer be alternative, but the verification step for web console method.

Reply:
IMO this step is just fine here....... It cannot be used for the web console version, because when you edit the YAML you would have already seen the values you have set there.

Reply:
@masauso-lungu I do not understand, the verification step says "on the {OCP} web console" so if it cannot be used for the web console, then it does not make sense..

Reply:
Otherwise if we are talking about the web console, we have to consider a different verification method  which does not involve reviewing the very YAML file you have edited.

Reply:
Oh yes, I understand that :) that is what I meant when I mentioned that the user sees if the values are added as they are adding it in web console, I do believe that is a good point you made. It just does not seem right to me to include this step in the cli procedure. But it is definitely not a big issue.

What I think is a bigger issue here, as I mentioned, we need to actually add some verification steps to the web console procedure. I do personally think that adding just verification steps that confirm that the values were added is not enough, even for the cli procedure. We should have some behavior that the user can observe in both cli and web console, for example that pod command that would check the number of pod replicas in the cluster, or the `argocd` command that I will update you about once I know if there is an alternative for downstream. I hope I explained myself better now :)  Thank you!